src/main/java/nextstep/subway/line/domain/Line.java

```diff
@@ -53,4 +53,48 @@ public Section addSection(Station upStation, Station downStation, Long distance)
         addSection(section);
         return section;
     }
+
+    public void removeStationInSection(Station station) {
+        removeStation(findPossibleStation(station));
+    }
+
+    private Station findPossibleStation(Station station) {
+        if (sections.getSections().size() <= 1) {
+            throw new IllegalStateException("구간에서 역을 제거할 수 없습니다.");
+        }
+
+        return sections.getStations().stream()
+                .filter(it -> it == station)
+                .findFirst()
+                .orElseThrow(() -> new IllegalArgumentException("제거 할 역이 노선 구간에 없습니다."));
+    }
+
+    private void removeStation(Station removePossibleStation) {
+        Optional<Section> upLineStation = getUpLineStation(removePossibleStation);
+        Optional<Section> downLineStation = getDownLineStation(removePossibleStation);
+
+        if (upLineStation.isPresent() && downLineStation.isPresent()) {
+            addNewSection(upLineStation.get(), downLineStation.get());
+        }
+
+        upLineStation.ifPresent(it -> sections.remove(it));
+        downLineStation.ifPresent(it -> sections.remove(it));
```

Comment:
개인적인 의견인데 up, down을 구분하지 않고 `List<Section>`과 같이 추출해 사용하면 어떨까요?

```suggestion
        List<Station> stations = getRelatedStations(removePossibleStation);

        if (stations.size == 2) {
            addNewSection(upLineStation.get(), downLineStation.get());
        }

        stations.forEach(it -> sections.remove(it));
```

Reply:
아하! 이렇게도 사용할 수 있겠군요..!
여러가지의 방향성을 제시해주셔서 너무 감사합니다 :)
규남님이 제시해주신 코드가 사실 보기엔 더 깔끔해 보이네요!! 피드백 감사드립니다👍😁